packages/playground/data-liberation/src/import/WP_Topological_Sorter.php

```diff
@@ -212,111 +249,243 @@ public function delete_session( $session_id ) {
+				'1.0.0'
+			);
 
-		if ( empty( $data ) ) {
 			return false;
 		}
 
-		$category_parent = null;
-
-		if ( array_key_exists( 'parent', $data ) && '' !== $data['parent'] ) {
-			$category_parent = $data['parent'];
-		}
+		return $this->get_mapped_element( $types[ $current_filter ], $data, $id, $additional_id );
+	}
 
-		$wpdb->insert(
-			self::get_table_name(),
-			array(
-				'session_id'   => $this->current_session,
-				'element_type' => self::ELEMENT_TYPE_CATEGORY,
-				'element_id'   => (string) $data['term_id'],
-				'parent_id'    => $category_parent,
-				'byte_offset'  => $byte_offset,
-				// Items with a parent has at least a sort order of 2.
-				'sort_order'   => $category_parent ? 2 : 1,
-			)
+	/**
+	 * Called by 'wxr_importer_processed_*' actions. This adds the entity to the
+	 * sorter table.
+	 *
+	 * @param int|null $id The ID of the element.
+	 * @param array $data The data to map.
+	 * @param int|null $additional_id The additional ID of the element.
+	 */
+	public function action_wxr_importer_processed( $id, $data, $additional_id = null ) {
+		$current_filter = current_action();
+		$types          = array(
+			'wxr_importer_processed_comment'      => 'comment',
+			'wxr_importer_processed_comment_meta' => 'comment_meta',
+			'wxr_importer_processed_post'         => 'post',
+			'wxr_importer_processed_post_meta'    => 'post_meta',
+			'wxr_importer_processed_term'         => 'term',
 		);
 
-		++$this->total_categories;
+		if ( ! $current_filter || ! array_key_exists( $current_filter, $types ) ) {
+			_doing_it_wrong(
+				__METHOD__,
+				'This method should be called by the wxr_importer_processed_* filters.',
+				'1.0.0'
+			);
+
+			return false;
+		}
+
+		$this->map_element( $types[ $current_filter ], $data, $id, $additional_id );
 	}
 
 	/**
-	 * Map a post to the index.
+	 * Map an element to the index. If $id is provided, it will be used to map the element.
 	 *
-	 * @param int $byte_offset The byte offset of the post.
-	 * @param array $data The post data.
+	 * @param string   $element_type The type of the element.
+	 * @param array    $data The data to map.
+	 * @param int|null $id The ID of the element.
+	 * @param int|null $additional_id The additional ID of the element.
 	 */
-	public function map_post( $byte_offset, $data ) {
+	public function map_element( $element_type, $data, $id = null, $additional_id = null ) {
 		global $wpdb;
 
-		if ( empty( $data ) ) {
-			return false;
+		if ( ! array_key_exists( $element_type, self::ENTITY_TYPES ) ) {
+			return;
 		}
 
-		// No parent, no need to sort.
-		if ( ! isset( $data['post_type'] ) ) {
-			return false;
+		$new_element = array(
+			'session_id'   => $this->current_session,
+			'element_type' => self::ENTITY_TYPES[ $element_type ],
+			'element_id'   => null,
+			'mapped_id'    => is_null( $id ) ? null : (string) $id,
+			'parent_id'    => null,
+			'byte_offset'  => 0,
+			// Items with a parent has at least a sort order of 2.
```

Comment:
Can we have a paragraph of text or a few to explain the idea behind `sort_order`?